### PR TITLE
trilinos: add scorec variant

### PR DESCRIPTION
**Pull request by bryanherman**
This PR adds SCOREC variant to Trilinos. 

https://redmine.scorec.rpi.edu/projects/albany-rpi/wiki/Installing_Trilinos/25

### CODE REVIEW

File: var/spack/repos/builtin/packages/trilinos/package.py [hunk 1]
```diff
@@ -599,6 +613,11 @@ def define_prefix_enable(prefix, cmake_var, spec_var=None):
                 define('Trilinos_ENABLE_SEACASNemslice', False)
             ])
 
+        if '+scorec' in spec:
+            options.extend([
+                define('Trilinos_ENABLE_SCOREC', True),
+            ])
+
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/trilinos/package.py]: Can you roll this into the long list at line 579?
```python
            define_trilinos_enable('SCOREC'),
```

The "special cases" down at the bottom here are for having to define nonstandard options.

**Review by sethrj — COMMENTED** [review]
Looks good aside from the one simplifying suggestion.

File: var/spack/repos/builtin/packages/trilinos/package.py (hunk 1 above)
Comment (bryanherman) [reply var/spack/repos/builtin/packages/trilinos/package.py]: Thanks for pointing that out, I missed that.

**Review by sethrj — APPROVED** [review]
Looks good, everyone loves more trilinos variants! 😂 Hopefully @keitat and the team can help with a standalone package once Trilinos' build system is modernized.

**Pull request merged**